sycl/test/basic_tests/esimd/flat_block.cpp

```diff
@@ -0,0 +1,20 @@
+// RUN: %clangxx -fsycl -fsycl-explicit-simd -fsycl-device-only -fsyntax-only -Xclang -verify %s
+// expected-no-diagnostics
+
+#include <CL/sycl.hpp>
+#include <limits>
+#include <sycl_esimd.hpp>
+#include <utility>
+
+using namespace sycl::intel::gpu;
+using namespace cl::sycl;
+
+void kernel(accessor<int, 1, access::mode::read_write, access::target::global_buffer> &buf) __attribute__((sycl_device)) {
+  simd<int, 32> v1(0, 1);
+
+  auto v0 = flat_block_load<int, 32>(buf.get_pointer());
```

Comment:
Should this `block_load` now?

Reply:
"flat" prefix is used whenever a memory API accespts a USM pointer, to distinguish from APIs which work with accessors. block_load would mean accessor-based API.

Reply:
Ah, I hadn't realized that.

Do you think this distinction is necessary?  If the only difference is whether the function accepts a pointer or an accessor, why not rely on overloading?  The fact that "flat" means "pointer" isn't obvious.

Reply:
I don't mind overloading, if that is better from interface stand point. @kychendev - do you have preference?

Reply:
I'd be interested in @rolandschulz's view here, too.  The C++ core guidelines suggest to [overload operations that are roughly equivalent](https://isocpp.github.io/CppCoreGuidelines/CppCoreGuidelines#Ro-equivalent), which I think applies here.

Sorry for not bringing this up sooner.  When reviewing the specification draft, I incorrectly assumed that `flat_load` simply meant `gather`, and didn't appreciate that there would still be a distinction between `gather` and `flat_gather`.

Reply:
Even though it looks similar, we have memory API targeting different contexts, such as access through SVM, SLM, or accessor. In core library implementation, we need to distinguish between different message types, address alignment, etc. and provide library developer the choice to produce the most efficient code. At high level, I'm fine with overloading, as long as the name is expressive and it can cover variations agreed by common users.

Reply:
You can distinguish between global pointers vs accessors and local pointers vs local accessors using just overloading, so I think that part is fine.

I agree that the different message types, address alignment and other parts of the API need more design work.  I'm happy if we add that to the TODO list and spend some more time discussing it.

Reply:
Renamed `flat_block_load` to `block_load`. Plus renamed tests.